src/frontend/lib/utils.ts

```diff
@@ -54,23 +54,20 @@ const GOOD_PRECISION = 10; // 10 meters
 export type LocationStatus = 'searching' | 'improving' | 'good' | 'error';
 
 export function getLocationStatus({
-  location,
-  providerStatus,
+  locationState,
 }: {
-  location?: LocationObject;
-  providerStatus?: LocationProviderStatus;
+  locationState?: LocationState;
 }): LocationStatus {
-  const gpsAvailable = !!providerStatus?.gpsAvailable;
-  const locationServicesEnabled = !!providerStatus?.locationServicesEnabled;
-
-  if (!gpsAvailable || !locationServicesEnabled) return 'error';
+  if (!locationState?.location || locationState.error) return 'error';
 
   const positionStale =
-    location && Date.now() - location.timestamp > STALE_TIMEOUT;
+    locationState &&
+    Date.now() - locationState.location.timestamp > STALE_TIMEOUT;
 
-  if (positionStale) return 'searching';
+  if (positionStale || !locationState.location.coords.accuracy)
```

Comment:
nit-pick: accuracy being `null` is not an indication that the GPS is searching, it is `null` if the accuracy is unavailable (which is only on web I think, so hence this is only a nit-pick)

Reply:
https://github.com/digidem/comapeo-mobile/pull/387/commits/0a7a4e5551fc0876d50c6d2675cb82c274c702b7